lib/slop/options.rb

```diff
@@ -88,7 +86,8 @@ def each(&block)
     #   o.on_error {|options, exception| puts exception; puts options; exit }
     #
     def on_error(&block)
-      @on_error_block = block
+      config[:on_error] = block
+      options.each{|o| o.on_error = block }
```

Comment:
I don't think this is necessary, is it? The parent config should be passed into all of the options when they're created

Reply:
But the order is important - the options created before the `on_error` call won't receive the `on_error` block. The test will fail without this iteration.